calrissian/job.py

```diff
@@ -212,6 +212,54 @@ def __init__(self, name, container_image, environment, volume_mounts, volumes, c
         self.security_context = security_context
         self.serviceaccount = serviceaccount
 
+        # For workflow steps, only mount the aws user-service volume
+        # And mount it to the shared credentials location
+        if self.name not in ["node_stage_in", "node_stage_out"]:
+            count = 0
+            for vol_m in self.volume_mounts[:]:
+                log.info(vol_m["name"])
+                log.info(vol_m["mountPath"])
+                if vol_m["name"].startswith("aws-credentials-service-"):
+                    vol_m["mountPath"] = vol_m["mountPath"].replace("/service", "")
+                    log.info("Updated mount path for workspace credentials")
+                    count += 1
+                if vol_m["name"].startswith("aws-credentials-workspace-"):
+                    self.volume_mounts.remove(vol_m)
+                    log.info("Removed aws volume for calling workspace")
+                    count += 1
+                if vol_m["name"].startswith("temp-pvc-workspace-"):
+                    self.volume_mounts.remove(vol_m)
+                    log.info("Removed volume for calling workspace")
+                    count += 1
+                if count == 3:
+                    break
```

Comment:
Are there sufficient extra volumes for this break logic to be worthwhile? It seems extra for a small gain to me, but not necessarily bad.